src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java

```diff
@@ -291,6 +292,79 @@ private static DetailAST getNextCurNode(DetailAST curNode) {
         return nodeToVisit;
     }
 
+    /**
+     * Checks whether the given node is next to an access specifier.
+     *
+     * @param parentNode the wrapped annotation.
+     * @return true if this node appears after an access specifier.
+     */
+    private static boolean isPreviousToPreviousSiblingAccessModifier(final DetailAST parentNode) {
+        return parentNode.getPreviousSibling() != null
+            && parentNode.getPreviousSibling().getPreviousSibling() != null
+            && Set.of(TokenTypes.LITERAL_PUBLIC, TokenTypes.LITERAL_PRIVATE,
+                    TokenTypes.LITERAL_PROTECTED)
+                .contains(parentNode.getPreviousSibling().getPreviousSibling().getType());
```

Comment:
This avoids redundant method calls:
```suggestion
        return Optional.ofNullable(parentNode.previousSibling)
                .map(DetailAst::getPreviousSibling)
                .filter(sibling -> TokenUtil.isOfType(sibling, TokenTypes.LITERAL_PUBLIC, TokenTypes.LITERAL_PRIVATE,
                    TokenTypes.LITERAL_PROTECTED))
                .isPresent();
```